src/CLIMA-atmos/Parameterizations/SurfaceFluxes/src/SurfaceFluxes.jl

```diff
@@ -0,0 +1,145 @@
+  return ψ_h
+end
+ψ_m_stable(ζ::R, ζ_0::R) where R = -PlanetParameters.beta_m * (ζ - ζ_0)
+ψ_h_stable(ζ::R, ζ_0::R) where R = -PlanetParameters.beta_h * (ζ - ζ_0)
+
+"""
+Computes roots of equation (eq. 10 in Ref. 1)
+
+  windspeed = u_* ( ln(z/z_0) - ψ_m(z/Λ, z_0/Λ) ) /κ        eq. 10 in Ref. 1
+  Λ = -u_*^3/(buoyancy_flux κ)                              eq. 22 in Ref. 2
+  ψ_m(ζ, ζ_0) = [ψ_m_stable, ψ_m_unstable]                  eq. 12, 14 in Ref 1
+
+Ref 1: Byun, Daewon W. "On the analytical solutions of flux-profile
+       relationships for the atmospheric surface layer." Journal of
+       Applied Meteorology 29.7 (1990): 652-657.
+       https://journals.ametsoc.org/doi/pdf/10.1175/1520-0450%281990%29029%3C0652%3AOTASOF%3E2.0.CO%3B2
+
+Ref 2: Wyngaard, John C. "Modeling the planetary boundary layer-Extension
+       to the stable case." Boundary-Layer Meteorology 9.4 (1975): 441-460.
+       https://sci-hub.tw/https://link.springer.com/article/10.1007/BF00223393
+"""
+function compute_friction_velocity(windspeed::R,
+                                   buoyancy_flux::R,
+                                   z_0::R,
+                                   z_1::R,
+                                   tol_abs::R,
+                                   iter_max::Int
+                                   )::R where R
+  # use neutral condition as first guess
+  logz = log(z_1 / z_0)
+  κ = PlanetParameters.kappa_vk
+  ustar_0 = windspeed * κ / logz
+  ustar = ustar_0
+
+  compute_Λ_MO(u) = - u * u * u / (buoyancy_flux * κ)
+  fixed_cond = z_1/compute_Λ_MO(ustar_0) >= 0.0
+  fixed_compute_ψ_m(ζ, ζ_0) = fixed_cond ? ψ_m_stable(ζ, ζ_0) : ψ_m_unstable(ζ, ζ_0)
+
+  function compute_ψ_m(u)
+    Λ_MO = compute_Λ_MO(u)
+    ζ   = z_1/Λ_MO
+    ζ_0 = z_0/Λ_MO
+    return fixed_compute_ψ_m(ζ, ζ_0)
+  end
+
+  compute_u_star_roots(u) = windspeed - u / κ * (logz - compute_ψ_m(u))
+  compute_ustar(u) = windspeed * κ / (logz - compute_ψ_m(u))
+
+  if (abs(buoyancy_flux) > 1.0e-20)
+    ustar_1 = compute_ustar(ustar_0)
+    args = ()
+    ustar, converged = RootSolvers.find_zero(compute_u_star_roots,
+                                             ustar_0, ustar_1,
+                                             args,
+                                             IterParams(tol_abs, iter_max),
+                                             SecantMethod()
+                                             )
+  end
+  return ustar
+end
+
+"""
+Ref 1: Monin-Obukhov similarity. Daewon W. Byun, 1990: On the
+       Analytical Solutions of Flux-Profile Relationships for
+       the Atmospheric Surface Layer. J. Appl. Meteor., 29, 652–657.
+       http://dx.doi.org/10.1175/1520-0450(1990)029<0652:OTASOF>2.0.CO;2
+"""
+function exchange_coefficients_byun(Ri::R, z_b::R, z_0::R)::Tuple{R,R,R} where R
+  γ_m = PlanetParameters.gamma_m
+  γ_h = PlanetParameters.gamma_h
+  β_h = PlanetParameters.beta_h
+  β_m = PlanetParameters.beta_m
+  κ = PlanetParameters.kappa_vk
```

Comment:
As in PlanetParameters, make these adjustable parameters that the DA/ML layer can adjust. So they need to passed into these functions in some parameter structure.